scanpy/neighbors/__init__.py

```diff
@@ -692,11 +837,11 @@ def compute_neighbors(
         if n_neighbors > self._adata.shape[0]:  # very small datasets
             n_neighbors = 1 + int(0.5*self._adata.shape[0])
             logg.warning(f'n_obs too small: adjusting to `n_neighbors = {n_neighbors}`')
-        if method == 'umap' and not knn:
+        if method in {'umap', 'tsne'} and not knn:
             raise ValueError('`method = \'umap\' only with `knn = True`.')
```

Comment:
This ValueError should now probably print `method=\'{method}\'` and not `method=\'umap\'`.